disk/disk.go

```diff
@@ -0,0 +1,90 @@
+package disk
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	diskLabel     = "var-lib-etcd"
+	maxRetries    = 15
+	retryInterval = time.Second * 10
+)
+
+var supportedFsType = []string{"ext4"}
+
+func WaitForDeviceReady(deviceName string) error {
+	b := backoff.NewMaxRetries(maxRetries, retryInterval)
+	o := func() error {
+		_, err := os.Stat(deviceName)
+		if os.IsNotExist(err) {
+			fmt.Printf("Waiting until device %q is registered by kernel.\n", deviceName)
+			return err
+		}
+		return nil
+	}
+	err := backoff.Retry(o, b)
+	if err != nil {
+		fmt.Printf("wait limit exceeded for device %q after %d retries\n", deviceName, maxRetries)
+		return microerror.Mask(err)
+	}
+	return nil
+}
+
+func MaybeCreateDiskFileSystem(deviceName string, desiredFsType string) error {
```

Comment:
```suggestion
func EnsureDiskHasFileSystem(deviceName string, desiredFsType string) error {
```

`Maybe` sounds slightly suspicious to me and doesn't clearly communicate what function would really do. WDYT about `Ensure`?